### ISSUE TITLE
Similarity Calculation Question

### ISSUE THREAD

**Issue by huang-xx** [mlfoundations/open_clip#541]
In the following code from README. Why multiply by `100` before calculating the softmax function? Instead of using `model.logit_scale`?

```python
import torch
from PIL import Image
import open_clip

model, _, preprocess = open_clip.create_model_and_transforms('ViT-B-32', pretrained='laion2b_s34b_b79k')
tokenizer = open_clip.get_tokenizer('ViT-B-32')

image = preprocess(Image.open("CLIP.png")).unsqueeze(0)
text = tokenizer(["a diagram", "a dog", "a cat"])

with torch.no_grad(), torch.cuda.amp.autocast():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)

    text_probs = (100.0 * image_features @ text_features.T).softmax(dim=-1)

print("Label probs:", text_probs)  # prints: [[1., 0., 0.]]

**Comment by mitchellnw** [comment]
This is the factor `np.exp(t)` in the code in Figure 3 of the CLIP paper (https://arxiv.org/pdf/2103.00020.pdf).

This is always 100 at the end of training (this is discussed elsewhere in the paper).

**Comment by huang-xx** [comment]
@mitchellnw Thanks, I got it.

In the original [implementation (https://github.com/openai/CLIP/blob/a9b1bf5920416aaeaec965c25dd9e8f98c864f16/clip/model.py#LL295C14-L295C25),   `t` is initialized as `np.log(1/0.07)`  and the logit is computed as `logit_scale = self.logit_scale.exp()`,  `logits_per_image = logit_scale * image_features @ text_features.t()`;  So, the `logit_scale` value is  `1 / 0.07` (around `14.29`) when the training begins.

According to the CLIP paper:

> The learnable temperature parameter τ was initialized to the equivalent of 0.07 from (Wu et al., 2018) and clipped to prevent scaling the logits by more than 100 which we found necessary to prevent training instability.  (page 5)

So, in your [train](https://github.com/mlfoundations/open_clip/blob/fb72f4db1b17133befd6c67c9cf32a533b85a321/src/training/train.py#L177) code,  the `logit_scale` is clamp to `4.6052 = ln(100)`  ;

My following question is, when computing the similarity between image-text tuples, is it correct to just calculate the cosine similarity between the normalized image_features and text_features?

```python
import torch.nn.functional as F
with torch.no_grad(), torch.cuda.amp.autocast():
    image_features = model.encode_image(image)
    text_features = model.encode_text(text)
    image_features /= image_features.norm(dim=-1, keepdim=True)
    text_features /= text_features.norm(dim=-1, keepdim=True)
    similarity_score = F.cosine_similarity(image_features, text_features, dim=-1)
